src/test/ui/closures/2229_closure_analysis/migrations/multi_diagnostics.stderr

```diff
@@ -90,8 +90,8 @@ LL |         let _f_1 = f1.1;
 LL | }
    | -
    | |
-   | in Rust 2018, `f1` would be dropped here, but in Rust 2021, only `f1.0` would be dropped here alongside the closure
    | in Rust 2018, `f1` would be dropped here, but in Rust 2021, only `f1.1` would be dropped here alongside the closure
+   | in Rust 2018, `f1` would be dropped here, but in Rust 2021, only `f1.0` would be dropped here alongside the closure
```

Comment:
I don't want to prevent this great work from landing, but I'm surprised by this change here. It looks like some slight non-determinism in the ordering of diagnostics. That can happen if they are emitted in an order that depends on iterating a HashMap/Set for example. 

If it's not fixed this PR can fail on CI while tests pass locally, or vice-versa, or remain hidden until it shows up and makes an unrelated PR fail CI.

Reply:
Good point, I think I know where the problem is. I'll work on a fix

Reply:
Should be fixed now

Reply:
Nice